### PR TITLE
#64 add init credentials

### LINKED ISSUE #64: Replace JWT with basic auth
Replace JWT with basic authorisation:

- Each request contains `authorization` header.
- Basic authorisation using `Bearer Base64(rego:secret)`.

### PR DESCRIPTION
**Pull request by wlthomson**
Fixes #64.

Branched off #66.

### CODE REVIEW

File: server/package.json [hunk 1]
```diff
@@ -37,7 +39,7 @@
     "eslint-plugin-react": "^6.10.3",
     "install": "^0.10.4",
     "nodemon": "^1.19.4",
-    "npm": "^6.14.9"
+    "typescript": "^4.1.3"
```
Comment (nmadruga) [inline server/package.json:42]: Uhh lala. Look at him go : D

File: server/postgres/initsqldb;C/init.sql [hunk 2]
```diff
@@ -0,0 +1,26 @@
+CREATE TABLE "site" (
```
Comment (nmadruga) [inline server/postgres/initsqldb;C/init.sql:1]: Is this file supposed to be here? I think it's `server/postgres/initsqldb/init.sql` all over again

File: server/postgres/queries.sql [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/* System */
+SELECT data->>'OS' as "os" FROM sites WHERE "UUID"='$site_uuid';
```
Comment (nmadruga) [inline server/postgres/queries.sql:2]: Hmm I'm curious about what is this for

Comment (nmadruga) [reply server/postgres/queries.sql:2]: this whole file

File: server/src/api/v1/api-doc.json [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+                "error": {
+                  "$ref": "#/definitions/Error"
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "/site/{siteId}/{machineId}": {
```
Comment (nmadruga) [inline server/src/api/v1/api-doc.json:121]: Ahá! Here it is, not I understand why: https://github.com/sussol/msupply/pull/7744#discussion_r557024132

File: README.md [hunk 5]
```diff
@@ -1,17 +1,14 @@
...
 Install node dependencies:
 
 `npm install`
 
-The hub database is run from within a docker container. To start the container:
+Start the postgres container:
 
 ```
-cd ./pgSql
+cd ./postgres
```
Comment (nmadruga) [inline README.md:11]: This is not clear that you need to be inside the `/server` folder.

File: README.md [hunk 6]
```diff
@@ -1,17 +1,14 @@
...
 ./pg-start.sh -r
```
Comment (nmadruga) [inline README.md:12]: Problem here:
```
Nicole:postgres nicole$ ./pg-start.sh -r
bash: ./pg-start.sh: Permission denied
```

I know how to fix, just editing the permission. But I think this should already come with correct permission. I'll try to clone again see if that will solve this problem

Comment (nmadruga) [reply README.md:12]: Screen Shot 2021-01-14 at 5 11 58 PM

File: server/config.json [hunk 7]
```diff
@@ -1,8 +1,10 @@
...
   "dev": {
     "port": 4000,
-    "secret": "secret",
+    "initSecret": "initSecret",
+    "tokenSecret": "tokenSecret",
     "bodyLimit": "100kb",
+    "logFormat": "dev",
     "logFileName": "./output.log",
     "pg": {
       "host": "localhost",
```
Comment (nmadruga) [inline server/config.json:10]: Seems that you just have one or the other now in the same server, so maybe just keep one here as well

**Review by nmadruga — COMMENTED** [review]
Half way through, will continue this tomorrow (and test). I had problems when trying to run the `pg-start.sh` script due to permissions. I know that I can fix that, but just a concern if that's how it comes when I clone the repo.

File: server/postgres/queries.sql (hunk 3 above)
Comment (wlthomson) [reply server/postgres/queries.sql:2]: These are just queries for the front-end. Once the dashboard is up and running, will probably replace them with some Grafana panel definitions!

File: README.md (hunk 5 above)
Comment (wlthomson) [reply README.md:11]: Oops, good point!

File: README.md (hunk 6 above)
Comment (wlthomson) [reply README.md:12]: Yeah, this script is from the original repo. I think it just needs a `chmod +x`. Will add to the `README.md`.

File: server/config.json (hunk 7 above)
Comment (wlthomson) [reply server/config.json:10]: Sorry, I don't follow! Are you asking why there are two secrets?

File: server/package.json (hunk 1 above)
Comment (wlthomson) [reply server/package.json:42]: 😄 .

File: server/postgres/initsqldb;C/init.sql (hunk 2 above)
Comment (wlthomson) [reply server/postgres/initsqldb;C/init.sql:1]: Yeah, it's just the same file but needs to be in this directory to work on Windows!

File: README.md (hunk 5 above)
Comment (wlthomson) [reply README.md:11]: I think at this point the `/server` directory is redundant. Suggest a new issue to flatten the repo to put all the code into the base directory.

**Pull request merged**